wherescape/connectors/gitlab/python_gitlab_create_metadata.py

```diff
@@ -0,0 +1,80 @@
+from datetime import datetime
+import logging
+
+from . import Gitlab
+from ...connectors.gitlab.gitlab_data_types_column_names import (
+    COLUMN_NAMES_AND_DATA_TYPES,
+)
+from ... import WhereScape
+from ...helper_functions import (
+    prepare_metadata_query,
+    create_column_names,
+    create_display_names,
+)
+
+
+def gitlab_create_metadata_project():
+    wherescape_instance = WhereScape()
+    access_token = wherescape_instance.read_parameter("gitlab_access_token")
+    base_url = wherescape_instance.read_parameter("gitlab_base_url")
+    gitlab_instance = Gitlab(access_token, base_url, "")
+    for project in gitlab_instance.projects:
+        logging.info(project[1])
+
+
+def gitlab_create_all_metadata():
+    wherescape_instance = WhereScape()
+
+    for resource_type in COLUMN_NAMES_AND_DATA_TYPES:
+        for key, value in resource_type:
+            gitlab_create_metadata(
+                wherescape_instance,
+                list(value.keys()),
+                list(value.values()),
+                key,
+            )
+
+
+def gitlab_create_metadata(wherescape_instance, columns, python_types, title):
+    start_time = datetime.now()
+    logging.info(
+        "Start time: %s for Gitlab_load_data" % start_time.strftime("%Y-%m-%d %H:%M:%S")
+    )
+    wherescape_object_id = wherescape_instance.object_key
+    types = []
+    for python_type in python_types:
+        if python_type == "int":
+            types.append("numeric")
+        elif python_type == "object":
+            types.append("text")
+        elif python_type == "datetime64[ns]":
+            types.append("timestamp")
+        elif python_type == "boolean":
+            types.append("boolean")
+
+    # Get the Gitlab database (definition)
+    logging.info(f'Fetched database definition "{title}" from Gitlab')
+
+    # Prepare the sql
+    column_names = create_column_names(columns)
+    display_names = create_display_names(columns)
+    sql = prepare_metadata_query(
+        wherescape_object_id,
+        "Gitlab api - " + title,
+        columns=column_names,
+        display_names=display_names,
+        types=types,
+        source_columns=columns,
+    )
```

Comment:
does the gitLab APi also prvide column descriptions?

Reply:
Unfortunately not!